ml-workflow-demo-kfp-katib-mlflow/README.md

```diff
@@ -0,0 +1,472 @@
+                          {
+                            "secretRef": {
+                              "name": "mlpipeline-minio-artifact"
+                            }
+                          }
+                        ]
+                    }
+                ]
+            }
+        }
+    }
+    tfjob_worker_spec = {
+        "replicas": 1,
+        "restartPolicy": "OnFailure",
+        "template": {
+            "metadata": {
+                "annotations": {
+                    "sidecar.istio.io/inject": "false"
+                }
+            },
+            "spec": {
+                "containers": [
+                    {
+                        "name": "tensorflow",
+                        "image": "docker.io/misohu/kubeflow-training:latest",
+                        "command": [
+                            "python",
+                            "/opt/model.py",
+                            "--s3-storage=true",
+                            f"{tf_model}",
+                            f"--bucket={bucket}",
+                            f"--bucket-key={key}",
+                        ],
+                        "envFrom": [
+                          {
+                            "secretRef": {
+                              "name": "mlpipeline-minio-artifact"
+                            }
+                          }
+                        ]
+                    }
+                ]
+            }
+        }
+    }
+    
+    # Create the KFP task for the TFJob.
+    tfjob_launcher_op = components.load_component_from_url(
+"https://raw.githubusercontent.com/kubeflow/pipelines/master/components/kubeflow/launcher/component.yaml")
+    op = tfjob_launcher_op(
+        name=tfjob_name,
+        namespace=tfjob_namespace,
+        chief_spec=json.dumps(tfjob_chief_spec),
+        worker_spec=json.dumps(tfjob_worker_spec),
+        tfjob_timeout_minutes=60,
+        delete_finished_tfjob=False)
+    return op
+```
+
+10. Define a complete pipeline that consists of all steps created earlier. Note that the name of the pipeline must be uqinue. If there was previously defined pipeline with the same name and within the same namespace either change the name of current pipeline or delete the older pipeline from the namespace.
+
+
+```python
+name = "demo-pipeline"
+namespace = "admin"
+s3_bucket = "demo-dataset"
+key = "data.csv"
+dataset_url = "https://www.openml.org/data/download/53995/KDDCup09_churn.arff"
+
+@dsl.pipeline(
+    name = "ML Workflow in Kubeflow",
+    description = "Demo pipeline"
+)
+def demo_pipeline(name=name, namespace=namespace):
+
+    # Step 1: Download dataset.
+    ingest_data_task = ingest_data_op(url=dataset_url)
+
+    # Step 2: Clean up the dataset and store it in S3 bucket.
+    clean_data_op = components.create_component_from_func(
+            clean_arff_data,
+            "clean_data.yaml",
+            packages_to_install=["pandas==1.2.4", "scipy==1.7.0", "boto3"],
+        )
```

Comment:
I like the idea of committing a `clean_data.yaml` file to the repo just so it is easy for people to see.  Just a suggestion